README.adoc

```diff
@@ -67,6 +67,14 @@ The actual port being used can then be retrieved by using `@LocalRunningGrpcPort
     enableReflection: true
 ----
 
+* Optionally set how long to wait for preexisting calls to finish during graceful server shutdown. New calls will be rejected during this time. A negative value is equivalent to an infinite grace period. Default value is `0`.
```

Comment:
Default value is `0` **(means `don't wait`)**

Reply:
> gave it a shot in this commit: 1776931. Let me know if that wasn't what you meant

Yes, but please provide all possible options : `-1`, `0`, `1` to cover the code.

> Also, if you have any advice on how to unit test or manually test this change, I'd love to verify it before merging.

I saw you already [able ](https://github.com/grpc/grpc-java/issues/6511)to manually reproduce and test the behavior. Lets relay on `grpc-java` team and skip the unit test for this  ;-)

Reply:
https://github.com/LogNet/grpc-spring-boot-starter/pull/139/commits/342c9452d1a9bbede5c762651c908a3496e5ef9b

Reply:
@jvmlet Is the `GRpcServerRunner.destroy` code path hit in the tests?

Reply:
Sorry, I overlooked the  `GRpcServerRunner.this.server.awaitTermination();`  in `GRpcServerRunner::startDaemonAwaitThread`.  It's better to move the await logic there

Reply:
>  It's better to move the await logic there

From my limited understanding of Springboot, the purpose of `startDaemonAwaitThread` is to keep the Springboot application running so that the gRPC server can serve requests, which is different than handling shutdown requests due to SIGINT or SIGTERM.

When the JVM is instructed to shutdown (such as via a SIGINT), it will call all registered shutdown handlers that are registered via [`addShutdownHook`](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#addShutdownHook-java.lang.Thread-):

```
Runtime.getRuntime().addShutdownHook(new Thread() {
    @Override
    public void run() {
        ...
    }
});
```

Once all those `run` methods return, JVM will shutdown.

I imagine that Springboot must add a JVM shutdown hook, and in that hook it will, through whatever internal Bean management mechanism, call the `destroy()` method on the Bean. This means that whenever the `destroy()` method returns, Springboot's shutdown hook will also return, and the JVM shuts down.

If all these assumptions are correct, then it is important to add an additional call to `server.awaitTermination()` within the `destroy()` method, so that we can give preexisting RPC calls time to finish before allowing the JVM to shut down.